scipy/special/_basic.py

```diff
@@ -2395,14 +2395,15 @@ def factorial2(n, exact=False):
     This is the factorial with every second value skipped.  E.g., ``7!! = 7 * 5
     * 3 * 1``.  It can be approximated numerically as::
 
-      n!! = special.gamma(n/2+1)*2**((m+1)/2)/sqrt(pi)  n odd
-          = 2**(n/2) * (n/2)!                           n even
+      n!! = 2 ** ((1 + 2*n - cos(pi * n))/4) * pi ** ((cos(pi * n) -1 )/4)
+            * gamma(1 + 0.5*n)
 
     Parameters
     ----------
-    n : int or array_like
-        Calculate ``n!!``.  Arrays are only supported with `exact` set
-        to False.  If ``n < 0``, the return value is 0.
+    n : float, complex, int or array_like
```

Comment:
Minor: I think we can just use `array_like` here.

From: https://numpydoc.readthedocs.io/en/latest/format.html

> array_like : For functions that take arguments which can have not only a type ndarray, but also types that can be converted to an ndarray (i.e. scalar types, sequence types), those arguments can be documented with type array_like.

I only learned that myself recently.